### PR TITLE
Fix -Wimplicit-function-declaration

### PR DESCRIPTION
**Pull request by thesamesam**
Causes build failures with Clang 16.

Bug: https://bugs.gentoo.org/875527
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

**Comment by thesamesam** [comment]
@sjaeckel could you possibly take a look? thanks

File: pb_add.c [hunk 1]
```diff
@@ -10,14 +10,15 @@
  * Tom St Denis, tomstdenis@iahu.ca, http://poly.libtomcrypt.org
  */
 #include <tompoly.h>
+#include <math.h>
```
Comment (sjaeckel) [inline pb_add.c]: Please apply the following patch to `tompoly.h` instead and use `PB_MIN()` resp. `PB_MAX()`

```
diff --git a/tompoly.h b/tompoly.h
index 646a903..cf79b69 100644
--- a/tompoly.h
+++ b/tompoly.h
@@ -114,2 +114,11 @@ int pb_readraw(pb_poly *a, unsigned char *buf, int len);
 
+/* What follows should be in a private header, but it's fine for now like that. */
+
+#ifndef PB_MIN
+#define PB_MIN(x, y) (((x) < (y)) ? (x) : (y))
+#endif
+#ifndef PB_MAX
+#define PB_MAX(x, y) (((x) > (y)) ? (x) : (y))
+#endif
+
 #endif
```

`used` is an `int` not a `float`, so using `fmin()` resp. `fmax()` doesn't really make sense.

`math.h` is also a pretty fat header file which isn't necessarily available on all platforms.

**Review by sjaeckel — CHANGES_REQUESTED** [review]
Sorry, I completely missed this PR!

PR looks fine besides this one remark.

File: pb_add.c (hunk 1 above)
Comment (thesamesam) [reply pb_add.c]: No worries! I did audit all the uses of the MAX/MIN here and a float should be fine for what they are, but I agree it's still a bit weird, and the macro is easier. I'll switch to that now, thanks!

Comment (thesamesam) [reply pb_add.c]: Flipped it over. Thanks!

**Review by sjaeckel — APPROVED** [review]

**Comment by thesamesam** [comment]
Thank you!

**Pull request merged**